Add support for similar waste types with different API IDs

EyeDevelop opened the pull request:
# Proposed Changes

This PR adds a fall-through method to the `WasteType` enum. The Twente Milieu API might return a waste type that is almost the same as one already defined in the enum.

Hengelo now has pick-up points for packaging and paper for high-density living. For the packaging, they created a new waste type. However, it is still packaging. This change now maps the 'odd' waste type returned from the API to the packages value in the enum.

Please let me know what you think!

## Related Issues

https://github.com/frenck/python-twentemilieu/issues/1265

frenck reviewed (changes requested):
Hi there @EyeDevelop 👋 

CI is failing, could you please take a look?

../Frenck

EyeDevelop commented:
Hi @frenck, I (should) have resolved all the issues with CI. Can you take a look?

I like the idea of having strict coding guidelines enforced with these typing rules and test coverage. However, if I could make a suggestion it would be to loosen them in some cases. For the `_missing` function I added to the `IntEnum`, mypy made a note that `value` should be of type `object`. However, in practice this can never be anything other than `int` unless the `_missing_` function is directly called. You can see in the test I added that I have to call the method directly to comply with the 100% coverage. It is my opinion that these are not really useful additions, but I leave this up to you.

frenck commented on .devcontainer/devcontainer.json:
```diff
@@ -50,5 +50,5 @@
   "image": "mcr.microsoft.com/devcontainers/python:3",
   "name": "Asynchronous Python client for Twente Milieu API",
-  "updateContentCommand": ". ${NVM_DIR}/nvm.sh && nvm install && nvm use && npm install && poetry install && poetry run pre-commit install"
+  "updateContentCommand": "git config --global --add safe.directory '*' && . ${NVM_DIR}/nvm.sh && nvm install && nvm use && npm install && poetry install && poetry run pre-commit install"
```
This change seems unrelated to the actual PR goal/description.

EyeDevelop replied:
This is a change I needed to get up to speed with the dev container you added. I can undo the commit if you'd prefer.

frenck commented on src/twentemilieu/twentemilieu.py:
```diff
@@ -34,6 +34,27 @@ class WasteType(IntEnum):
+
+        return {
+            56: WasteType.PACKAGES,
+        }[int(value)]
```
The cast seems unneeded here, as the instance is already validated above.

frenck commented on tests/test_twentemilieu.py:
```diff
@@ -40,6 +40,18 @@ async def test_json_request(aresponses: ResponsesMockServer) -> None:
+    assert WasteType(56) == WasteType.PACKAGES, \
+        "Fallback for high-density packages not handled!"
+
+    try:
+        WasteType._missing_("wrong_type")
+        pytest.fail("Should not be able to call the fallback with invalid type.")
+    except TypeError:
+        pass
```
Use `pytest.raises` instead, avoiding the need to add try/catch constructs.

frenck reviewed (changes requested):
> However, if I could make a suggestion it would be to loosen them in some cases.

Sorry... I strongly disagree very much. Don't know what else to say.

Left a few review comments and the CI is also failing (which is odd? As ruff is run in pre-commit... not sure how you manage to make a commit that passes it).

../Frenck

EyeDevelop commented:
Processed your review comments. Let me know if I should drop the first commit with the changes to the dev container file.

frenck reviewed (approved):
Thanks, @EyeDevelop 👍 

../Frenck
